tests/fast_int.cpp

```diff
@@ -0,0 +1,379 @@
+    unsigned result;
+    auto answer = fast_float::from_chars(f.data(), f.data() + f.size(), result, 2);
+    if (answer.ec != std::errc::invalid_argument) {
+      std::cerr << "expected error should be 'invalid_argument' for: " << std::quoted(f) << std::endl;
+      return EXIT_FAILURE;
+    }
+  }
+
+  // octal test
+  const std::vector<int> base_octal_test_expected {0, 1, 7, 8, 9};
+  const std::vector<std::string> base_octal_test { "0", "1", "07", "010", "0011" };
+
+  for (std::size_t i = 0; i < base_octal_test.size(); ++i)
+  {
+    const auto& f = base_octal_test[i];
+    int result;
+    auto answer = fast_float::from_chars(f.data(), f.data() + f.size(), result, 8);
+    if (answer.ec != std::errc()) {
+      std::cerr << "could not convert to int for input: " << std::quoted(f) << std::endl;
+      return EXIT_FAILURE;
+    }
+    else if (result != base_octal_test_expected[i]) {
+      std::cerr << "result "  << std::quoted(f) << " did not match with expected int: " << base_octal_test_expected[i] << std::endl;
+      return EXIT_FAILURE;
+    }
+  }
+
+  // hex test
+  const std::vector<int> base_hex_test_expected { 0, 1, 15, 16, 0, 16};
+  const std::vector<std::string> base_hex_test { "0", "1", "F", "01f", "0x11", "10X11" };
+
+  for (std::size_t i = 0; i < base_hex_test.size(); ++i)
+  {
+    const auto& f = base_hex_test[i];
+    int result;
+    auto answer = fast_float::from_chars(f.data(), f.data() + f.size(), result, 16);
+    if (answer.ec != std::errc()) {
+      std::cerr << "could not convert to int for input: " << std::quoted(f) << std::endl;
+      return EXIT_FAILURE;
+    }
+    else if (result != base_hex_test_expected[i]) {
+      std::cerr << "result "  << std::quoted(f) << " did not match with expected int: " << base_hex_test_expected[i] << std::endl;
+      return EXIT_FAILURE;
+    }
+  }
+
+  // invalid base test (-1) 
+  const std::vector<std::string> invalid_base_test { "0", "1", "-1", "F", "10Z" };
+
+  for (std::size_t i = 0; i < invalid_base_test.size(); ++i)
+  {
+    const auto& f = invalid_base_test[i];
+    int result;
+    auto answer = fast_float::from_chars(f.data(), f.data() + f.size(), result, -1);
+    if (answer.ec != std::errc::invalid_argument) {
+      std::cerr << "expected error should be 'invalid_argument' for: " << std::quoted(f) << std::endl;
+      return EXIT_FAILURE;
+    }
+  }
+
+  // out of range base test (100)
+  const std::vector<int> base_out_of_range_test_expected { 0, 1, 15, 35, 10035 };
+  const std::vector<std::string> base_out_of_range_test { "0", "1", "F", "Z", "10Z" };
+
+  for (std::size_t i = 0; i < base_out_of_range_test.size(); ++i)
+  {
+    const auto& f = base_out_of_range_test[i];
+    int result;
+    auto answer = fast_float::from_chars(f.data(), f.data() + f.size(), result, 100);
+    if (answer.ec != std::errc()) {
+      std::cerr << "could not convert to int for input: " << std::quoted(f) << std::endl;
```

Comment:
invalid bases are always rejected with std::errc::invalid_argument, so this test will never succeed.